xc/device_common.ml

```diff
@@ -266,6 +270,25 @@ let list_frontends ~xs ?for_devids domid =
                                ) devids)
                  ) kinds)
 
+(* Network SR-IOV VF device *)
+let make_nsv_device ?(back_domid=0) front_domid devid =
+  let frontend = { domid = front_domid; kind = NetSriovVf; devid = devid } in
+  let backend = { domid = back_domid; kind = NetSriovVf; devid = devid } in
+  { backend = backend; frontend = frontend }
+
+let list_xenserver_devices ~xs ?for_devids domid =
+  let xenserver_dir = sprintf "/local/domain/%d/xenserver/device" domid in
+  let mk = function
```

Comment:
This doesn't feel right. We should list the directory, then look at the paths to determine the device kinds and make devices for them. We don't need to have special cases to create different kinds. I don't think this is inherently different from what `list_frontends` does already: only the `frontend_dir` is different, isn't it?